lib/gc.cpp

```diff
@@ -86,6 +81,51 @@ void *operator new(std::size_t size) {
     return rv;
 }
 
+void *operator new(std::size_t size, const std::nothrow_t &) noexcept {
+    TRACE_ALLOC(size)
+
+    maybe_initialize_gc();
+    // FIXME: Call nothrow operator new from libgc with suitable new libgc
+    // versions
+    auto *rv = ::operator new(size, UseGC, 0, 0);
+    if (!rv && emergency_ptr && emergency_ptr + size < emergency_pool + sizeof(emergency_pool)) {
+        rv = emergency_ptr;
+        size += -size & 0xf;  // align to 16 bytes
```

Comment:
Not your code but this particular piece looks a little funky to me.  I had to pause a bit to think what negating an unsigned means and what this is trying to do. Testing for the right behavior in case of a bad alloc seems tricky.

Reply:
You'd just need to get used to calculate masks on-fly :) I changed this thing, is it more clear now?